File: Rio/Rio.Web/Modules/ServerTypes/Workspace/GroupStudentForm.ts
import { LookupEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface GroupStudentForm {
    GroupId: LookupEditor;
    StudentId: LookupEditor;
}

export class GroupStudentForm extends PrefixedContext {
    static formKey = 'Workspace.GroupStudent';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!GroupStudentForm.init)  {
            GroupStudentForm.init = true;

            var w0 = LookupEditor;

            initFormType(GroupStudentForm, [
                'GroupId', w0,
                'StudentId', w0
            ]);
        }
    }
}

File: Rio/Rio.Web/Modules/Workspace/GroupStudent/GroupStudentForm.cs
using Serenity.ComponentModel;
using Serenity.Web;
using System;

namespace Rio.Workspace.Forms
{
    [FormScript("Workspace.GroupStudent")]
    [BasedOnRow(typeof(GroupStudentRow), CheckNames = true)]
    public class GroupStudentForm
    {
        [HalfWidth]
        public int GroupId { get; set; }
        [HalfWidth]
        public long StudentId { get; set; }
        [HalfWidth]
        public int TeacherId { get; set; }
    }
}